hw/ip/tlul/adapter_dmi.core

```diff
@@ -0,0 +1,65 @@
+CAPI=2:
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+name: "lowrisc:tlul:adapter_dmi:0.1"
+description: "TL-UL to DMI interface adapter"
+
+filesets:
+  files_rtl:
+    depend:
+      - lowrisc:prim:assert
+      - lowrisc:prim:secded
+      - lowrisc:tlul:common
+      - lowrisc:tlul:trans_intg
+      - pulp-platform:riscv-dbg:0.1
```

Comment:
I'm seeing quite a strange issue. Pulling in `pulp-platform:riscv-dbg:0.1` causes the `rv_dm` (and some other IPs) smoke tests to fail. Already the build step.

```
                Error-[SV-LCM-PND] Package not defined
                src/lowrisc_dv_rv_dm_env_0.1/rv_dm_env_pkg.sv, 21
                rv_dm_env_pkg, "rv_dm_reg_pkg::NrHarts"
                  Package scope resolution failed. Token 'rv_dm_reg_pkg' is not a package.
                  Originating module 'rv_dm_env_pkg'.
 ``` 

I am really puzzled what the issue is here.

Reply:
Ah, that's a bug in `rv_dm_env.core`. I think it should depend on `rv_dm` but instead it depends on the debug module directly. Presumably this causes a problem when you add the extra dependency (which "shakes the tree" a bit and the arbitrary ordering no longer gets what we need).

That should be fixed by #24763 (I hope!)

Reply:
Right, that's merged. If you rebase, I'm hoping it should be fixed :-)

Reply:
Awesome. Thanks man!